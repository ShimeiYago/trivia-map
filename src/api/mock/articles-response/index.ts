import { GetArticlesPreviewsResponse } from 'api/articles-api/get-articles-previews';
import { GetMyArticlesResponse } from 'api/articles-api/get-my-articles';
import { GetArticleResponse } from 'api/articles-api/get-remote-article';
import { PostArticleResponse } from 'api/articles-api/post-remote-article';

export const mockGetArticleResponse: GetArticleResponse = {
  postId: 1,
  title: 'ノーチラス号',
  description: 'ネモ船長が最大の発明だと自負する潜水艇。',
  marker: {
    markerId: 1,
    lat: -111.33,
    lng: 149.243,
    park: 'S',
    numberOfPublicArticles: { total: 4, eachCategory: [1, 2, 1] },
    areaNames: ['シー', 'メディテレーニアンハーバー', 'ポルトパラディーゾ'],
  },
  image: 'https://disneyparkstory.com/wp-content/uploads/2017/02/CIMG9800.jpg',
  isDraft: false,
  author: {
    userId: 1,
    nickname: 'Axel',
    icon: 'https://disneyparkstory.com/wp-content/uploads/2017/09/apple-touch-icon.png',
  },
  createdAt: '2022/06/16 21:42',
  updatedAt: '2022/06/16 21:42',
  category: 1,
  numberOfLikes: 1,
};

export const mockPostArticleResponse: PostArticleResponse = {
  postId: 1,
  title: 'ノーチラス号',
  description: 'ネモ船長が最大の発明だと自負する潜水艇。',
  marker: 100,
  image: null,
  isDraft: false,
  author: 0,
  category: 1,
  createdAt: '2022/06/16 21:42',
  updatedAt: '2022/06/16 21:42',
};

export const mockGetArticlesPreviewsResponse: GetArticlesPreviewsResponse = {
  nextUrl: null,
  previousUrl: null,
  totalRecords: 2,
  totalPages: 2,
  currentPage: 1,
  startIndex: 1,
  endIndex: 1,
  results: [
    {
      postId: 1,
      title: 'ノーチラス号のエンジン',
      image: 'https://disneyparkstory.com/wp-content/uploads/2017/02/CIMG9800.jpg',
      category: 1,
      createdAt: '2022/06/16 21:42',
      numberOfLikes: 1,
    },
    {
      postId: 2,
      title: 'ノーチラス号の名前の由来',
      image: null,
      category: 1,
      createdAt: '2022/06/16 21:42',
      numberOfLikes: 1,
    },
  ],
};

export const mockGetMyArticlesResponse: GetMyArticlesResponse = {
  nextUrl: null,
  previousUrl: null,
  totalRecords: 2,
  totalPages: 2,
  startIndex: 1,
  endIndex: 1,
  currentPage: 1,
  results: [
    {
      postId: 1,
      title: 'ノーチラス号のエンジン',
      isDraft: false,
      category: 1,
    },
    {
      postId: 2,
      title: 'ノーチラス号の名前の由来',
      isDraft: true,
      category: 1,
    },
  ],
};
