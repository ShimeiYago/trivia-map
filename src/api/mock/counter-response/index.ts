import { GetCounterResponse, PostCounterResponse } from '../../counter-api';

export const getCounterMockResponse: GetCounterResponse = {
  count: 5,
};

export const postCounterMockResponse: PostCounterResponse = {
  count: 5,
};
