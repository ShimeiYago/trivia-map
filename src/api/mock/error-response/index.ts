import { AxiosError } from 'axios';

export const axiosMockError = {
  response: {
    status: 500,
    data: {},
  },
  message: 'Intentional API Error with mock',
} as AxiosError;
