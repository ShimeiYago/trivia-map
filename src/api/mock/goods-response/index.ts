import { ToggleGoodResponse } from '../../goods-api/toggle-good';
import { CheckGoodStatusResponse } from 'api/goods-api/check-good-status';

export const mockCheckGoodStatusResponse: CheckGoodStatusResponse = {
  haveAddedGood: true,
};

export const mockToggleGoodResponse: ToggleGoodResponse = {
  haveAddedGood: true,
  numberOfGoods: 2,
};
