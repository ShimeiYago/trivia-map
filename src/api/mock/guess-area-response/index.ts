import { GuessAreaResponse } from '../../guess-area';

export const mockGuessAreaResponse: GuessAreaResponse = {
  areaNames: ['シー', 'メディテレーニアンハーバー', 'ポルトパラディーゾ'],
};
