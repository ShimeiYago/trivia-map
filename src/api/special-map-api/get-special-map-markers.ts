import { PaginationResponse } from '../types/pagination-response';
import { BASE_URL } from 'constant';
import { AxiosError, AxiosResponse } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';
import { mockGetSpecialMapMarkersResponseWithPagination } from 'api/mock/special-map-response';
import { getUrlParameters } from 'utils/get-url-parameters';
import { Park } from 'types/park';
import { MapMarkerVariant } from 'types/marker-icon';

export async function getSpecialMapMarkers(param: {
  mapId: number;
  page?: number;
}): Promise<GetSpecialMapMarkersResponseWithPagination> {
  const urlParams = getUrlParameters({ page: param.page });
  const url = `${BASE_URL}/special-map/maps/${param.mapId}/markers${urlParams}`;

  const axiosInstance = getAxiosInstance({}, mockGetSpecialMapMarkersResponseWithPagination);

  try {
    const res: AxiosResponse<GetSpecialMapMarkersResponseWithPagination> = await axiosInstance.get(
      url,
    );
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError(axiosError);
  }
}

export type GetSpecialMapMarkersResponse = {
  specialMapMarkerId: number;
  lat: number;
  lng: number;
  park: Park;
  image: string | null;
  description: string;
  variant: MapMarkerVariant;
};

export type GetSpecialMapMarkersResponseWithPagination =
  PaginationResponse<GetSpecialMapMarkersResponse>;
