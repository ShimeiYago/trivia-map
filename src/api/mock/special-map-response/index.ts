import { GetSpecialMapResponse } from 'api/special-map-api/get-special-map';
import { GetSpecialMapMarkersResponseWithPagination } from 'api/special-map-api/get-special-map-markers';
import { GetSpecialMapsResponseWithPagination } from 'api/special-map-api/get-special-maps';

export const mockGetSpecialMapResponse: GetSpecialMapResponse = {
  specialMapId: 1,
  title: 'トイレマップ',
  thumbnail: 'https://www.yuu-diaryblog.com/wp-content/uploads/2017/06/disney-wc.jpg',
  isPublic: true,
  description: '定番から穴場まで、ディズニーのトイレマップです。',
  selectablePark: 'both',
};

export const mockGetSpecialMapsResponseWithPagination: GetSpecialMapsResponseWithPagination = {
  nextUrl: null,
  previousUrl: null,
  totalRecords: 2,
  totalPages: 2,
  currentPage: 1,
  startIndex: 1,
  endIndex: 1,
  results: [
    mockGetSpecialMapResponse,
    {
      specialMapId: 2,
      title: '「関係者以外立ち入り禁止」看板マップ',
      thumbnail: null,
      isPublic: true,
      description: 'パーク中の「関係者以外立ち入り禁止」の看板のまとめです。',
      selectablePark: 'L',
    },
  ],
};

export const mockGetSpecialMapMarkersResponseWithPagination: GetSpecialMapMarkersResponseWithPagination =
  {
    nextUrl: null,
    previousUrl: null,
    totalRecords: 2,
    totalPages: 1,
    currentPage: 1,
    startIndex: 1,
    endIndex: 1,
    results: [
      {
        specialMapMarkerId: 1,
        lat: -67.6875,
        lng: 140.75,
        park: 'L',
        image:
          'https://cdn-ak.f.st-hatena.com/images/fotolife/n/nats-co/20160323/20160323164220.jpg',
        description: 'アリスモチーフのトイレです。トランプ模様の扉がかわいい。',
        variant: 'blue',
      },
      {
        specialMapMarkerId: 2,
        lat: -202.34375,
        lng: 163.1875,
        park: 'S',
        image: null,
        description: 'エントランスのトイレならここ。',
        variant: 'blue',
      },
    ],
  };
