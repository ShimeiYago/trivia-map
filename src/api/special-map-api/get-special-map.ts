import { BASE_URL } from 'constant';
import { AxiosError, AxiosResponse } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';
import { mockGetSpecialMapResponse } from 'api/mock/special-map-response';
import { SelectablePark } from 'types/park';
import { Author } from 'types/author';

export async function getSpecialMap(mapId: number): Promise<GetSpecialMapResponse> {
  const url = `${BASE_URL}/special-map/maps/${mapId}/detail`;

  const axiosInstance = getAxiosInstance({}, mockGetSpecialMapResponse);

  try {
    const res: AxiosResponse<GetSpecialMapResponse> = await axiosInstance.get(url);
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError(axiosError);
  }
}

export type GetSpecialMapResponse = {
  specialMapId: number;
  author: Author;
  title: string;
  thumbnail: string | null;
  isPublic: boolean;
  description: string;
  selectablePark: SelectablePark;
  minLatitude: number;
  maxLatitude: number;
  minLongitude: number;
  maxLongitude: number;
  createdAt: string;
};
