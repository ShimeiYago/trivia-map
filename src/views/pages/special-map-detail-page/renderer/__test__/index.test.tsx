import { shallow, ShallowWrapper } from 'enzyme';
import { Renderer, State, Props } from '..';
import * as GetSpecialMapApiModule from 'api/special-map-api/get-special-map';
import * as GetSpecialMapMarkersApiModule from 'api/special-map-api/get-special-map-markers';
import { mockGetSpecialMapResponse } from 'api/mock/special-map-response';
import { mockGetSpecialMapMarkersResponseWithPagination } from 'api/mock/special-map-response';

let shallowWrapper: ShallowWrapper<Props, State, Renderer>;
let getSpecialMapSpy: jest.SpyInstance;
let getSpecialMapMarkersSpy: jest.SpyInstance;

const props: Props = {
  throwError: jest.fn(),
  mapId: 0,
  isMobile: false,
};

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    shallowWrapper = shallow(<Renderer {...props} />);
  });

  it('basic', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('loading markers', () => {
    shallowWrapper.setState({
      loadingSpecialMap: false,
      specialMap: mockGetSpecialMapResponse,
      loadingMarkers: true,
    });
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('loaded', () => {
    shallowWrapper.setState({
      loadingSpecialMap: false,
      specialMap: mockGetSpecialMapResponse,
      loadingMarkers: false,
      markers: mockGetSpecialMapMarkersResponseWithPagination.results,
    });
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('loaded (without thumbnail)', () => {
    shallowWrapper.setState({
      loadingSpecialMap: false,
      specialMap: { ...mockGetSpecialMapResponse, thumbnail: null },
      loadingMarkers: false,
      markers: mockGetSpecialMapMarkersResponseWithPagination.results,
    });
    expect(shallowWrapper).toMatchSnapshot();
  });
});

describe('fetchSpecialMap', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    shallowWrapper = shallow(<Renderer {...props} />);
    getSpecialMapSpy = jest.spyOn(GetSpecialMapApiModule, 'getSpecialMap');
  });

  it('should set specialMapsResponseWithPagination if api calling succeed', async () => {
    getSpecialMapSpy.mockResolvedValue(mockGetSpecialMapResponse);

    const instance = shallowWrapper.instance();
    await instance['fetchSpecialMap']();

    expect(instance.state.specialMap?.isPublic).toBeTruthy();
  });

  it('should call throwError if api calling fail', async () => {
    getSpecialMapSpy.mockRejectedValue(new Error());

    const instance = shallowWrapper.instance();
    await instance['fetchSpecialMap']();

    expect(instance.props.throwError).toBeCalled();
  });

  it('should call throwError 404 when api fail with 404', async () => {
    getSpecialMapSpy.mockRejectedValue({ status: 404 });

    const instance = shallowWrapper.instance();
    await instance['fetchSpecialMap']();

    expect(instance.props.throwError).toHaveBeenCalledWith(404);
  });
});

describe('fetchSpecialMapMarkers', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    shallowWrapper = shallow(<Renderer {...props} />);
    getSpecialMapMarkersSpy = jest.spyOn(GetSpecialMapMarkersApiModule, 'getSpecialMapMarkers');
  });

  it('should set markers if api calling succeed', async () => {
    getSpecialMapMarkersSpy.mockResolvedValue({
      ...mockGetSpecialMapMarkersResponseWithPagination,
      totalPages: 2,
    });

    const instance = shallowWrapper.instance();
    await instance['fetchSpecialMapMarkers']();

    expect(instance.state.loadingMarkers).toBeFalsy();
  });

  it('should call throwError if api calling fail', async () => {
    getSpecialMapMarkersSpy.mockRejectedValue(new Error());

    const instance = shallowWrapper.instance();
    await instance['fetchSpecialMapMarkers']();

    expect(instance.props.throwError).toBeCalled();
  });
});
