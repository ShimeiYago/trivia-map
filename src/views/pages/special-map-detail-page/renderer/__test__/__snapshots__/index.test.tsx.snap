// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `
<ArticleWrapper
  localBackNavi={
    Object {
      "link": "/special-maps/0",
      "text": "全体マップへ戻る",
    }
  }
  showSidebar={true}
>
  <ForwardRef(Box)
    my={1}
  >
    <CenterSpinner />
  </ForwardRef(Box)>
</ArticleWrapper>
`;

exports[`Shallow Snapshot Tests loaded (without thumbnail) 1`] = `
<ArticleWrapper
  localBackNavi={
    Object {
      "link": "/special-maps/0",
      "text": "トイレマップへ戻る",
    }
  }
  showSidebar={true}
>
  <CommonHelmet
    description="定番から穴場まで、ディズニーのトイレマップです。"
    imageUrl="http://localhost/site-image-special-map.png"
    ogType="article"
    title="トイレマップ"
  />
  <ForwardRef(Stack)
    spacing={2}
  >
    <ShareButtons
      description="定番から穴場まで、ディズニーのトイレマップです。"
      title="トイレマップ"
      url="http://localhost/special-maps/1"
    />
    <ForwardRef(Divider) />
    <AuthorLink
      author={
        Object {
          "icon": "https://disneyparkstory.com/wp-content/uploads/2017/09/apple-touch-icon.png",
          "nickname": "Axel",
          "url": "https://disneyparkstory.com",
          "userId": 1,
        }
      }
    />
    <DynamicAlignedText
      component="h2"
      sx={
        Object {
          "wordBreak": "break-all",
        }
      }
      variant="h5"
    >
      トイレマップ
    </DynamicAlignedText>
    <DynamicAlignedText
      fontSize={18}
      whiteSpace="pre-wrap"
    >
      定番から穴場まで、ディズニーのトイレマップです。
    </DynamicAlignedText>
    <ForwardRef(Divider) />
    <IconAndText
      align="center"
      component="h3"
      iconComponent={
        <MyIcon
          variant="map-marker"
        />
      }
      iconPosition="left"
      text="ピン一覧"
      variant="h6"
    />
    <ForwardRef(Stack)
      spacing={5}
    >
      <ForwardRef(Box)
        bgcolor="#f5f8fa"
        border="3px solid #424242"
        key="special-map-marker--67.6875-140.75"
      >
        <NonStyleLink
          to="/special-maps/0?marker=1&park=L"
        >
          <ForwardRef(Box)
            height={200}
            mx="auto"
          >
            <ParkMap
              disabled={true}
              initCenter={
                Object {
                  "lat": -67.6875,
                  "lng": 140.75,
                }
              }
              initZoom={2}
              park="L"
            >
              <MapMarker
                autoOpen={false}
                draggable={false}
                position={
                  Object {
                    "lat": -67.6875,
                    "lng": 140.75,
                  }
                }
                variant="restroom"
                zIndexOffset={0}
              />
            </ParkMap>
          </ForwardRef(Box)>
        </NonStyleLink>
        <ForwardRef(Box)
          p={1}
        >
          <DynamicAlignedText
            component="div"
            mb={1}
            whiteSpace="pre-wrap"
          >
            アリスモチーフのトイレです。トランプ模様の扉がかわいい。
          </DynamicAlignedText>
          <ForwardRef(Typography)
            align="center"
            component="div"
            mb={1}
          >
            <Image
              maxHeight="300px"
              maxWidth="full"
              src="https://cdn-ak.f.st-hatena.com/images/fotolife/n/nats-co/20160323/20160323164220.jpg"
            />
          </ForwardRef(Typography)>
          <NonStyleLink
            to="/special-maps/0?marker=1&park=L"
          >
            <ForwardRef(Typography)
              align="right"
            >
              <ForwardRef(Button)>
                <IconAndText
                  component="span"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="地図上で確認する"
                />
              </ForwardRef(Button)>
            </ForwardRef(Typography)>
          </NonStyleLink>
        </ForwardRef(Box)>
      </ForwardRef(Box)>
      <ForwardRef(Box)
        bgcolor="#f5f8fa"
        border="3px solid #424242"
        key="special-map-marker--140.46875-111.0625"
      >
        <NonStyleLink
          to="/special-maps/0?marker=2&park=S"
        >
          <ForwardRef(Box)
            height={200}
            mx="auto"
          >
            <ParkMap
              disabled={true}
              initCenter={
                Object {
                  "lat": -140.46875,
                  "lng": 111.0625,
                }
              }
              initZoom={2}
              park="S"
            >
              <MapMarker
                autoOpen={false}
                draggable={false}
                position={
                  Object {
                    "lat": -140.46875,
                    "lng": 111.0625,
                  }
                }
                variant="restroom"
                zIndexOffset={0}
              />
            </ParkMap>
          </ForwardRef(Box)>
        </NonStyleLink>
        <ForwardRef(Box)
          p={1}
        >
          <DynamicAlignedText
            component="div"
            mb={1}
            whiteSpace="pre-wrap"
          >
            エントランスのトイレならここ。
          </DynamicAlignedText>
          <NonStyleLink
            to="/special-maps/0?marker=2&park=S"
          >
            <ForwardRef(Typography)
              align="right"
            >
              <ForwardRef(Button)>
                <IconAndText
                  component="span"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="地図上で確認する"
                />
              </ForwardRef(Button)>
            </ForwardRef(Typography)>
          </NonStyleLink>
        </ForwardRef(Box)>
      </ForwardRef(Box)>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
</ArticleWrapper>
`;

exports[`Shallow Snapshot Tests loaded 1`] = `
<ArticleWrapper
  localBackNavi={
    Object {
      "link": "/special-maps/0",
      "text": "トイレマップへ戻る",
    }
  }
  showSidebar={true}
>
  <CommonHelmet
    description="定番から穴場まで、ディズニーのトイレマップです。"
    imageUrl="https://www.yuu-diaryblog.com/wp-content/uploads/2017/06/disney-wc.jpg"
    ogType="article"
    title="トイレマップ"
  />
  <ForwardRef(Stack)
    spacing={2}
  >
    <ShareButtons
      description="定番から穴場まで、ディズニーのトイレマップです。"
      title="トイレマップ"
      url="http://localhost/special-maps/1"
    />
    <ForwardRef(Divider) />
    <AuthorLink
      author={
        Object {
          "icon": "https://disneyparkstory.com/wp-content/uploads/2017/09/apple-touch-icon.png",
          "nickname": "Axel",
          "url": "https://disneyparkstory.com",
          "userId": 1,
        }
      }
    />
    <DynamicAlignedText
      component="h2"
      sx={
        Object {
          "wordBreak": "break-all",
        }
      }
      variant="h5"
    >
      トイレマップ
    </DynamicAlignedText>
    <Image
      src="https://www.yuu-diaryblog.com/wp-content/uploads/2017/06/disney-wc.jpg"
      width="full"
    />
    <DynamicAlignedText
      fontSize={18}
      whiteSpace="pre-wrap"
    >
      定番から穴場まで、ディズニーのトイレマップです。
    </DynamicAlignedText>
    <ForwardRef(Divider) />
    <IconAndText
      align="center"
      component="h3"
      iconComponent={
        <MyIcon
          variant="map-marker"
        />
      }
      iconPosition="left"
      text="ピン一覧"
      variant="h6"
    />
    <ForwardRef(Stack)
      spacing={5}
    >
      <ForwardRef(Box)
        bgcolor="#f5f8fa"
        border="3px solid #424242"
        key="special-map-marker--67.6875-140.75"
      >
        <NonStyleLink
          to="/special-maps/0?marker=1&park=L"
        >
          <ForwardRef(Box)
            height={200}
            mx="auto"
          >
            <ParkMap
              disabled={true}
              initCenter={
                Object {
                  "lat": -67.6875,
                  "lng": 140.75,
                }
              }
              initZoom={2}
              park="L"
            >
              <MapMarker
                autoOpen={false}
                draggable={false}
                position={
                  Object {
                    "lat": -67.6875,
                    "lng": 140.75,
                  }
                }
                variant="restroom"
                zIndexOffset={0}
              />
            </ParkMap>
          </ForwardRef(Box)>
        </NonStyleLink>
        <ForwardRef(Box)
          p={1}
        >
          <DynamicAlignedText
            component="div"
            mb={1}
            whiteSpace="pre-wrap"
          >
            アリスモチーフのトイレです。トランプ模様の扉がかわいい。
          </DynamicAlignedText>
          <ForwardRef(Typography)
            align="center"
            component="div"
            mb={1}
          >
            <Image
              maxHeight="300px"
              maxWidth="full"
              src="https://cdn-ak.f.st-hatena.com/images/fotolife/n/nats-co/20160323/20160323164220.jpg"
            />
          </ForwardRef(Typography)>
          <NonStyleLink
            to="/special-maps/0?marker=1&park=L"
          >
            <ForwardRef(Typography)
              align="right"
            >
              <ForwardRef(Button)>
                <IconAndText
                  component="span"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="地図上で確認する"
                />
              </ForwardRef(Button)>
            </ForwardRef(Typography)>
          </NonStyleLink>
        </ForwardRef(Box)>
      </ForwardRef(Box)>
      <ForwardRef(Box)
        bgcolor="#f5f8fa"
        border="3px solid #424242"
        key="special-map-marker--140.46875-111.0625"
      >
        <NonStyleLink
          to="/special-maps/0?marker=2&park=S"
        >
          <ForwardRef(Box)
            height={200}
            mx="auto"
          >
            <ParkMap
              disabled={true}
              initCenter={
                Object {
                  "lat": -140.46875,
                  "lng": 111.0625,
                }
              }
              initZoom={2}
              park="S"
            >
              <MapMarker
                autoOpen={false}
                draggable={false}
                position={
                  Object {
                    "lat": -140.46875,
                    "lng": 111.0625,
                  }
                }
                variant="restroom"
                zIndexOffset={0}
              />
            </ParkMap>
          </ForwardRef(Box)>
        </NonStyleLink>
        <ForwardRef(Box)
          p={1}
        >
          <DynamicAlignedText
            component="div"
            mb={1}
            whiteSpace="pre-wrap"
          >
            エントランスのトイレならここ。
          </DynamicAlignedText>
          <NonStyleLink
            to="/special-maps/0?marker=2&park=S"
          >
            <ForwardRef(Typography)
              align="right"
            >
              <ForwardRef(Button)>
                <IconAndText
                  component="span"
                  iconComponent={<Memo />}
                  iconPosition="left"
                  text="地図上で確認する"
                />
              </ForwardRef(Button)>
            </ForwardRef(Typography)>
          </NonStyleLink>
        </ForwardRef(Box)>
      </ForwardRef(Box)>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
</ArticleWrapper>
`;

exports[`Shallow Snapshot Tests loading markers 1`] = `
<ArticleWrapper
  localBackNavi={
    Object {
      "link": "/special-maps/0",
      "text": "トイレマップへ戻る",
    }
  }
  showSidebar={true}
>
  <CommonHelmet
    description="定番から穴場まで、ディズニーのトイレマップです。"
    imageUrl="https://www.yuu-diaryblog.com/wp-content/uploads/2017/06/disney-wc.jpg"
    ogType="article"
    title="トイレマップ"
  />
  <ForwardRef(Stack)
    spacing={2}
  >
    <ShareButtons
      description="定番から穴場まで、ディズニーのトイレマップです。"
      title="トイレマップ"
      url="http://localhost/special-maps/1"
    />
    <ForwardRef(Divider) />
    <AuthorLink
      author={
        Object {
          "icon": "https://disneyparkstory.com/wp-content/uploads/2017/09/apple-touch-icon.png",
          "nickname": "Axel",
          "url": "https://disneyparkstory.com",
          "userId": 1,
        }
      }
    />
    <DynamicAlignedText
      component="h2"
      sx={
        Object {
          "wordBreak": "break-all",
        }
      }
      variant="h5"
    >
      トイレマップ
    </DynamicAlignedText>
    <Image
      src="https://www.yuu-diaryblog.com/wp-content/uploads/2017/06/disney-wc.jpg"
      width="full"
    />
    <DynamicAlignedText
      fontSize={18}
      whiteSpace="pre-wrap"
    >
      定番から穴場まで、ディズニーのトイレマップです。
    </DynamicAlignedText>
    <ForwardRef(Divider) />
    <IconAndText
      align="center"
      component="h3"
      iconComponent={
        <MyIcon
          variant="map-marker"
        />
      }
      iconPosition="left"
      text="ピン一覧"
      variant="h6"
    />
    <ForwardRef(Stack)
      spacing={5}
    >
      <CenterSpinner />
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
</ArticleWrapper>
`;
